agnpy/absorption/absorption.py

```diff
@@ -479,6 +586,116 @@ def tau_blr_mu_s(self, nu):
+                self.r
+            )
+            )
+        return tau_blr_list
+
+    def tau_blr_all_lines_cubepy(self, nu, eps_abs=1e-6):
+        """
+        Calculate the absorption in all available BLR (Broad Line Region) shells.
+        This function scales the radius and luminosity to the Hbeta line
+        and applies the computation to all the available BLR lines (shells) using
+        their respective ratios.
+
+        Parameters
+        ----------
+        nu: numpy array
+            Frequency array with unit
+        eps_abs: float, optional
+            Absolute precision for the calculation, default is 1e-6
+        """
+        SUM_RATIO_LINES_BLR = 30.652
+        XI_TARGET_LINE = self.target.xi_line
+        tau_z_all = np.zeros(nu.shape)
+
+        blr_shells = dict()
+        absorption_shells = dict()
+
+        # TODO write test
+        if self.target.line != "Hbeta":
+            raise KeyError("Please use Hbeta as reference")
+
+        for line_key in lines_dictionary.keys():
+            xi_line_current = lines_dictionary[line_key]['L_Hbeta_ratio']
+            xi_line = (xi_line_current * XI_TARGET_LINE) / SUM_RATIO_LINES_BLR
+            R_line_current = self.target.R_line * lines_dictionary[line_key]['R_Hbeta_ratio']
+
+            blr_shells[line_key] = SphericalShellBLR(self.target.L_disk, xi_line, line_key, R_line_current)
+            absorption_shells[line_key] = Absorption(blr_shells[line_key], r=self.r, z=self.z)
+
+            tau_z_current = absorption_shells[line_key].tau_blr_cubepy(nu, eps_abs=eps_abs)
+            tau_z_all += tau_z_current
+
+        return tau_z_all
+
+    def tau_blr_lines_cubepy(self, nu, lines_list, eps_abs=1e-6):
+        """
+        Calculate the absorption in all available BLR (Broad Line Region) shells.
+        This function scales the radius and luminosity to the Hβ (H-beta) line
+        and computes the absorption for all available BLR lines (shells) using
+        their respective ratios.
+
+        Parameters
+        ----------
+        nu : numpy.ndarray
+            Frequency array for which the absorption will be calculated.
+        lines_list : list of str
+            List of BLR emission line names (e.g., 'Hbeta', 'MgII') for which the
+             absorption will be calculated. Each name corresponds to a predefined
+            line with associated properties (e.g., wavelength, radius ratio, and
+            luminosity ratio) that are stored as lines_dictionary.
+        eps_abs : float, optional
+            Absolute precision for the calculation (default is 1e-6).
+
+        Returns
+        -------
+        tau : numpy.ndarray
+             Optical depth values corresponding to the input frequencies.
+        """
+
+        sum_ratio_lines = sum(lines_dictionary[line]['L_Hbeta_ratio'] for line in lines_list)
+        XI_TARGET_LINE = self.target.xi_line
+        tau_z_lines = np.zeros(nu.shape)
+
+        blr_shells = dict()
```

Comment:
copying the comment from the previous function:



why do you define those dictionaries? I guess it is a remnant of a script where you were also plotting the effect of individual lines, but this is either way not possible this function.

here you are using the dictionaries only to sum up over all the lines, so you could instead created SphericalShellBLR and Absorption as temporary variables directly in the for loop